From: SeongJae Park <sj@kernel.org>
Date: Sun, 12 Jan 2025 11:12:23 -0800
Subject: [PATCH] mm/damon/core: unset damos->walk_completed after confimed set

damos->walk_completed is only set, not unset.  This can cause next
damos_walk() finish earlier than expected.  Unset it after all
walk_completed is confirmed.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 579662d6af99..a94b086d6d57 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -1517,6 +1517,9 @@ static void damos_walk_complete(struct damon_ctx *ctx, struct damos *s)
 		if (!siter->walk_completed)
 			return;
 	}
+	damon_for_each_scheme(siter, ctx)
+		siter->walk_completed = false;
+
 	complete(&control->completion);
 	mutex_lock(&ctx->walk_control_lock);
 	ctx->walk_control = NULL;
-- 
2.39.5

